surya/detection/__init__.py

```diff
@@ -68,12 +95,13 @@ def batch_detection(
             self,
             images: List,
             batch_size=None,
-            static_cache=False
+            static_cache=False,
+            inline=False
     ) -> Generator[Tuple[List[List[np.ndarray]], List[Tuple[int, int]]], None, None]:
         assert all([isinstance(image, Image.Image) for image in images])
         if batch_size is None:
             batch_size = self.get_batch_size()
-        heatmap_count = self.model.config.num_labels
+        heatmap_count = self.model['inline' if inline else 'text'].config.num_labels
```

Comment:
We should just select the right model with model = self.model['inline' if inline...], then use it vs copying the logic multiple times